url/tests/unit.rs

```diff
@@ -147,6 +147,17 @@ fn new_path_windows_fun() {
     }
 }
 
+#[test]
+fn set_scheme_with_leading_or_trailing_c0_controls_should_result_in_noop() {
+    let mut url: Url = "http://test".parse().unwrap();
+
+    let controls = ["\u{0000}", "\u{000C}", "\u{000E}", "\u{0020}"];
+    for control in controls.iter() {
+        assert!(url.set_scheme(&format!("{}https", control)).is_err());
```

Comment:
The web-platform-test says `Non-tab/newline C0 controls result in no-op` - but this returns an error. Is that intentional?
Could you update [urltestdata.json](https://github.com/servo/rust-url/blob/master/url/tests/urltestdata.json) with the testcases you listed [here](https://github.com/web-platform-tests/wpt/blob/f72c88ab79843a46ffdf63264e833121811f08d9/url/resources/setters_tests.json#L291-L323) ?

Reply:
> The web-platform-test says `Non-tab/newline C0 controls result in no-op` - but this returns an error. Is that intentional?

Sorry if I misunderstood but I thought `result in no-op` means the url scheme will not be updated but the operation of `set_scheme` failed due to invalid input for scheme so it returns an error here. Or, would it be better to return `Ok` here? 👀 

> Could you update [urltestdata.json](https://github.com/servo/rust-url/blob/master/url/tests/urltestdata.json) with the testcases you listed [here](https://github.com/web-platform-tests/wpt/blob/f72c88ab79843a46ffdf63264e833121811f08d9/url/resources/setters_tests.json#L291-L323) ?

Yes! I will update the urltestdata.json file later 🙏

Reply:
As the tests referred in the issue in under `setters_tests.json`, I just sent a separate PR to update the test cases at #820.

Reply:
After thinking more about the test, I think I was wrong because if it should return an error, the test would test `failure: true` instead. So, I just updated the PR to make it `no-op` 🙏 Please help me review again, thanks for catching it!